src/views/components/item.jsx

```diff
@@ -1,16 +1,56 @@
 import React from 'react';
-
+import ItemStore from '../../store/item-store.js';
+import { Link } from 'react-router';
 export default class Item extends React.Component {
+    constructor(props) {
+        super(props);
+        if (props.item !== null)
+            this.state = {item: props.item};
+        else
+            this.state = {item: null};
+    }
 
-    render() {
-        if (this.props.item === null) {
-            return <i>No item found</i>;
+    componentDidMount() {
+        if (this.state.item === null) {
+            ItemStore.searchItemByAddress(this.props.params.address).then(() => {
+                this.setState({
+                    item: ItemStore.getItem()
+                })
+            });
         }
+    }
+
+    render() {
+        if (this.state.item === null)
+            return (<i>Item is loading...</i>);
         return (
             <div className='item'>
-                Item ID: {this.props.item.id}
-                <br />
-                Item status: {this.props.item.status}
+                <div className="picArea">
+                    <img src="../assets/images/placeholder.jpg"/>
+                </div>
+                <div className="titleArea">
+                    <h2>{this.state.item.address}</h2>
+                    <Link to={`/model/${this.state.item.modelAddress}`}>View model</Link>
+                </div>
+                <div className="infoArea">
+                    <div className="descriptionArea">
+                        <h3>Status</h3>
+                        <p>{this.state.item.status}</p>
+                    </div>
+                    <div className="faultArea">
+                        <h3>Fault</h3>
+                        {(this.state.item.isFaulty
+                                ? <p>{this.state.item.faultDescription}</p>
+                                : <p>Item is not faulty.</p>
```

Comment:
Maybe italicize the "Item is not faulty." message to differentiate it from a fault message?